### DISCUSSION TITLE
C4GT: Discovering Tutors and Mentor using DSEP protocol

### DISCUSSION THREAD

**Discussion by cchaitanya27** [Sunbird-Ed/Community#490] [Feature request]
### Project Details
Beckn (short for Beckn protocol) is an open protocol for decentralized commerce. It consists of several specifications for building open e-commerce networks across which sellers and their offers are universally discoverable from any Beckn-enabled app or platform.
Several particularly famous implementations of Beckn protocol are [ONDC - Open Network Digital Commerce](https://ondc.org/), [Namma Yatri](https://nammayatri.in/), [Unified Health Interface](https://uhi.abdm.gov.in/)
This is to enable discovery of Tutors and Mentors that are registered in a DSEP network of Tutors/Mentors through Sunbird ED Application. Tutoring can be helpful for students who need additional support as it provides Individualized attention, Personalized learning. Online Tutoring is more flexible as it is easy for scheduling sessions at convenient times and locations. Moreover, it can make learning more engaging and interactive, with the use of multimedia resources, virtual whiteboards, and other digital tools. 

### Features to be implemented
Interface in the Sunbird App for the learner to discover published tutor / mentor service on a DSEP based mentor network as per the learner’s requirements
- When a learner scans a QR code in a book, the details of the book such as Grade, Medium, Subject linked to the QR code are provided. The app should search the registered tutor/mentor service based on this context.
- There is a menu option in the Sunbird App where it opens a search page to search and discover tutor/mentor service based on user profile

### What exists
Sunbird ED App
DSEP Protocol

### What needs to be built
Enhance Sunbird ED App to discover tutors/mentors in a DSEP based network

### Learning Path
Complexity: Medium
Skills Required: 
Understanding existing Sunbird ED App, DSEP Protocol, NodeJS, Angular
Mentors: Niharika, Chinmay
Project size: 6 weeks

### Milestones
Detailing and documenting the capabilities
Creating the design for integrating DSEP protocol in Sunbird Mobile App
Implementation of the changes to Sunbird Mobile App
Test cases

**Comment by chitranshu-keshav** [comment]
@Sunbird-Ed/sunbirded_workinggroup  Can you please review and provide your inputs on this feature request

**Comment by chunkomac** [comment]
Which discord channel should interested folks join for this project ?

**Reply by kameshbhr** [reply]
Diksha channel

**Comment by aks30** [comment]
For anyone interested in this project, a prototype on similar lines was done by the Shikshalokam team as part of the Skill-a-thon Hackathon organized by EkStep and Beckn. The prototype demonstrated how SunbirdEd (Diksha app) could look like when integrated with the DSEP protocol. More details here - https://github.com/skill-a-thon/.github/blob/main/OSL-challange-1.md

To others looking for a fresh implementation, a basic setup guide on how to leverage Mentoring BPP and enable the discovery of mentors/tutors on Diksha can refer to the [setup guide here](https://docs.google.com/document/d/1XjRuvDs_4_gGQv7ap07WNDFbC00Ulx1y8zNfp8hYh_4/edit?usp=sharing) which shall serve as a great starting point to build upon for this project.

Happy to help answer any queries or for further details do feel free to reach out.

Cc - @vijiurs @kiranharidas187 @kameshbhr

**Reply by bruno-noir** [reply]
Thank you for sharing the prototype and setup guide. I am excited working on the proposal and hope for the opportunity to contribute.:)
